### PR TITLE
Add TLS and CA Certificate support for wins

### PR DESCRIPTION
**Pull request by rosskirkpat**
add new functionality for fetching certificates from Rancher, external URL, and from a local file.

### CODE REVIEW

File: cmd/server/config/config.go [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
-	"os"
-	"strings"
-
 	"github.com/ghodss/yaml"
 	"github.com/pkg/errors"
 	"github.com/rancher/system-agent/pkg/config"
 	"github.com/rancher/wins/pkg/csiproxy"
 	"github.com/rancher/wins/pkg/defaults"
+	"github.com/rancher/wins/pkg/tls"
+	"io/ioutil"
```
Comment (luthermonson) [inline cmd/server/config/config.go:10]: goimports?

Comment (rosskirkpat) [reply cmd/server/config/config.go]: fixed now - goland was being clunky

**Comment by rosskirkpat** [comment]
https://github.com/rancher/rancher/issues/37483

**Review by phillipsj — CHANGES_REQUESTED** [review]
Can you please document this in the README.

File: pkg/tls/tls.go [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+const (
+	rancherCertPath      = "cacerts"
+	localRancherCertDir  = "C:\\etc\\rancher\\agent\\"
+	localRancherCertFile = localRancherCertDir + "ranchercert"
+	localGenericCertDir  = "C:\\certs\\"
+	localGenericCertFile = localGenericCertDir + "customcert"
+)
+
+func SetupRancherTLSConfig() (*x509.CertPool, error) {
+	if os.Getenv("CATTLE_CA_CHECKSUM") == "" {
```
[the remark this answers is not in the document]
Comment (rosskirkpat) [reply pkg/tls/tls.go]: This has been removed and I've simplified it down to a single function that is called in csiproxy if the config value is set for `certfilepath`

**Comment by rosskirkpat** [comment]
README updated and implementation in csiproxy added. Re-requesting reviews now.

**Review by phillipsj — APPROVED** [review]

**Review by luthermonson — APPROVED** [review]
Squash your commits tho

**Pull request merged**